protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
> We get `ErrBindingsDiffer` (either in verifyReturnedPromise or verifyAuthPath) iff msg.Error == ReqNameExisted, right?

No, not "iff".  If the already registered key is the one we expect, we'd get `ReqNameExisted` but `CheckPassed`.  Also, on request, if the server does the registration but sneaks in some other key, it would return `Success` but we'd get `ErrBindingsDiffer`.

> In other words, does ReqNameExisted always inconsistent with our expectation (assume that we never do registration twice for a user)?

Are you asking if finding a `ReqNameExisted` is always, in itself, unexpected?  Or just finding a differing key?  I think there're a lot of cases where this may come up.  For example, the client disconnects when registering.  The server may have received the requests but didn't have time to send a response.  At which point, the client would reconnect and try again.  It would then receive a `ReqNameExisted` but the bindings would match.  Another more common scenario might be, a user registers normally but then loses their device or deletes their local profile or sets up a new device, or in some other way legitimately generates a new key for their account.  They attempt to register and now they'd get `ReqNameExisted` but `ErrBindingsDiffer`.